### DISCUSSION TITLE
flux CLI does not see Kustomise betav1 after the upgrade?

### DISCUSSION THREAD

**Discussion by andrey-dubnik** [fluxcd/flux2#1962] [Q&A] [answered]
Hi, we have Flux upgraded on some of the clusters which now requires 2 CLIs to manage. Is this a bug or we need to rely on multiple CLI versions in future handling different APIs?

```
flux get kustomization --all-namespaces
```

> ✗ no matches for kind "Kustomization" in version "kustomize.toolkit.fluxcd.io/v1beta2"

**Comment by stefanprodan** [comment, marked as answer]
The CLI doesn't support multiple API versions, this is something we'll consider after the `v1` release of our APIs.

**Comment by kingdonb** [comment]
To clarify this, v1beta1 API for Kustomize controller is automatically upgraded to v1beta2 when your cluster controllers are upgraded. The old version `v1beta1` remains in etcd, but it should be made available upgraded as the new version `v1beta2` as soon as the bootstrap upgrade is completed.

So unless you are still running both versions of Flux from before and after the v1beta2 upgrade (in separate clusters, I suppose) then you should not need to have two versions of the CLI, only the latest one.

**Reply by andrey-dubnik** [reply]
Thanks, in my case I was running different flux version on the cluster while using the same cli hence it was not recognising v1beta1. Otherwise all looking good.
